201-sqlmi-new-vnet-w-point-to-site-vpn/scripts/deploy.ps1

```diff
@@ -10,16 +10,52 @@ $parameters.Remove('subscriptionId')
 $parameters.Remove('resourceGroupName')
 $parameters.Remove('certificateNamePrefix')
 
-function Ensure-Login () 
+$managedInstanceName = $parameters['managedInstanceName']
+
+function EnsureLogin() 
 {
     $context = Get-AzureRmContext
-    If($context.Subscription -eq $null)
+    If($null -eq $context.Subscription)
     {
         Login-AzureRmAccount | Out-null
     }
 }
 
-Ensure-Login
+function VerifyPSVersion
+{
+    Write-Host "Verifying PowerShell version, must be 5.0 or higher."
+    if($PSVersionTable.PSVersion.Major -ge 5)
+    {
+        Write-Host "PowerShell version verified." -ForegroundColor Green
+    }
+    else
+    {
+        Write-Host "You need to install PowerShell version 5.0 or heigher." -ForegroundColor Red
+        Break;
+    }
+}
+
+function VerifyManagedInstanceName
+{
+    param($managedInstanceName)
+    Write-Host "Verifying Managed Instance name, must be globally unique."
+    if([string]::IsNullOrEmpty($managedInstanceName))
+    {
+        Write-Host "Managed Instance name is required parameter." -ForegroundColor Red
+        break;
+    }
+    if($null -ne (Resolve-DnsName ($managedInstanceName+'.provisioning.database.windows.net') -ErrorAction SilentlyContinue))
```

Comment:
is this script still needed post GA?  If so, we need to update this check to not hard-code the uri

Reply:
Post GA it will be replaced by API call, for now this is the way to check